go/armel/Dockerfile.tmpl

```diff
@@ -39,11 +39,11 @@ RUN apt install -y \
 {{- if or (eq .DEBIAN_VERSION "10") (eq .DEBIAN_VERSION "11") (eq .DEBIAN_VERSION "12")}}
 # librpm-dev
 RUN apt install -y \
-        librpm-dev:armel
+        librpm-dev:armel libxml2:armel
 
 # libsystemd-dev
-RUN apt install -y \
-        libsystemd-dev:armel
+RUN apt-get install -y --allow-remove-essential \
```

Comment:
I'm not sure about this, can you elaborate what's the reason for doing so?

Reply:
This was done when I faced issue with upgrading package libsystemd0, you can check error in build that I will provide.
This is solution that I found now.
But Im not sure why we face this error only at debian 11. 

Example of error:
https://buildkite.com/elastic/golang-crossbuild/builds/561#0191dbad-1ea4-4a90-92ef-44558d5c86bf

Reply:
It is an error and breaks the cross compilation. You will remove packages from arch x86 and replace them with armel packages. You have to find the packages of arch armel you have to install. x86 and armel packages are not compatible.

Reply:
unmet dependencies is something that this project has faced a few times, once per Debian version:
- https://github.com/search?q=repo%3Aelastic%2Fgolang-crossbuild+unmet+dependencies&type=issues

Those issues might explain how things were fixed, in case it helps for fixing this issue

Reply:
Thank you for comments @v1v , @kuisathaverat 
I will procced to work with it now.

Reply:
> IMO, these four architectures should be supported only on the latest version of Debian

I agree too

Beats use the versions below:

https://github.com/elastic/beats/blob/3a5cd592d226a5d22e9927e45a7bfeb2a98b024f/dev-tools/mage/crossbuild.go#L222-L245

I guess that's a question for the Ingest team to answer.

Reply:
Those 4 platforms (the ones removed in https://github.com/elastic/golang-crossbuild/pull/456) are not in our support matrix and we do not publish builds for them.

There may be someone out there who uses this to build their own custom projects for Debian 10 and not 12, if they do we can direct them to open a PR adding support back in a non breaking way. I don't think we have the capacity to support every permutation here for the ones that aren't actually used in Elastic products anymore.

Reply:
+1 to removing them.